src/test/java/com/google/cloud/spanner/pgadapter/python/django/DjangoPgAggregates.java

```diff
@@ -0,0 +1,230 @@
+            + "  select 3802 as oid, 'jsonb' as typname, (select oid from pg_namespace where nspname='pg_catalog') as typnamespace, null as typowner, -1 as typlen, false as typbyval, 'b' as typtype, 'U' as typcategory, false as typispreferred, true as typisdefined, ',' as typdelim, 0 as typrelid, 0 as typelem, 3807 as typarray, 'jsonb_in' as typinput, 'jsonb_out' as typoutput, 'jsonb_recv' as typreceive, 'jsonb_send' as typsend, '-' as typmodin, '-' as typmodout, '-' as typanalyze, 'i' as typalign, 'x' as typstorage, false as typnotnull, 0 as typbasetype, -1 as typtypmod, 0 as typndims, 0 as typcollation, null as typdefaultbin, null as typdefault, null as typacl\n"
+            + ")\n"
+            + "SELECT typarray FROM pg_type WHERE typname = 'citext'";
+
+    List<String> result1 = new ArrayList<>();
+    result1.add("string");
+    result1.add("some_column");
+
+    ResultSet rs1 = createResultSet(result1, 1);
+
+    mockSpanner.putStatementResult(StatementResult.query(Statement.of(unnecessarySql), rs1));
+    mockSpanner.putStatementResult(StatementResult.query(Statement.of(unnecessarySql2), rs1));
+  }
+
+  @Test
+  public void strAggTest() throws IOException, InterruptedException {
+
+    String sql =
+        "SELECT \"singers\".\"firstname\", STRING_AGG(\"singers\".\"firstname\", '|' ) AS \"str\" FROM \"singers\" GROUP BY \"singers\".\"firstname\" LIMIT 21";
+
+    List<String> result = new ArrayList<>();
+
+    result.add("string");
+    result.add("firstname");
+
+    result.add("string");
+    result.add("str");
+
+    result.add("hello");
+    result.add("hello|world");
+
+    result.add("world");
+    result.add("world|hello");
+
+    mockSpanner.putStatementResult(
+        StatementResult.query(Statement.of(sql), createResultSet(result, 2)));
+
+    List<String> options = new ArrayList<>();
+    options.add("string_agg");
+
+    String actualOutput = executePgAggregateTests(pgServer.getLocalPort(), host, options);
+    String expectedOutput =
+        "<QuerySet [{'firstname': 'hello', 'str': 'hello|world'}, {'firstname': 'world', 'str': 'world|hello'}]>\n";
+
+    assertEquals(expectedOutput, actualOutput);
+
+    assertEquals(3, mockSpanner.countRequestsOfType(ExecuteSqlRequest.class));
+    assertEquals(sql, mockSpanner.getRequestsOfType(ExecuteSqlRequest.class).get(2).getSql());
+  }
+
+  @Test
+  public void arrAggTest() throws IOException, InterruptedException {
+
+    String sql = "SELECT ARRAY_AGG(\"singers\".\"firstname\" ) AS \"arr\" FROM \"singers\"";
```

Comment:
Here and elsewhere: This generates a valid PostgreSQL SQL statement, but is this function supported in Spangres?

Reply:
Yes, I first tested these on a sample database in the cloud spanner via PG Adapter and then only added the unit tests